examples/SampleApp/src/components/NewDirectMessagingSendButton.tsx

```diff
@@ -0,0 +1,182 @@
+  Us extends UnknownType = DefaultUserType
+> = Pick<
+  MessageInputContextValue<At, Ch, Co, Ev, Me, Re, Us>,
+  'giphyActive' | 'sendMessage'
+> & {
+  /** Disables the button */ disabled: boolean;
+};
+const SendButtonWithContext = <
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+>(
+  props: SendButtonPropsWithContext<At, Ch, Co, Ev, Me, Re, Us>,
+) => {
+  const { disabled = false, giphyActive, sendMessage } = props;
+  const {
+    theme: {
+      colors: { accent_blue, grey_gainsboro },
+      messageInput: { sendButton },
+    },
+  } = useTheme();
+  return (
+    <TouchableOpacity
+      disabled={disabled}
+      onPress={sendMessage}
+      style={[sendButton]}
+      testID='send-button'
+    >
+      {giphyActive && <Search pathFill={accent_blue} />}
+      {!giphyActive && disabled && <SendRight pathFill={grey_gainsboro} />}
+      {!giphyActive && !disabled && <SendUp pathFill={accent_blue} />}
+    </TouchableOpacity>
+  );
+};
+const areEqual = <
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+>(
+  prevProps: SendButtonPropsWithContext<At, Ch, Co, Ev, Me, Re, Us>,
+  nextProps: SendButtonPropsWithContext<At, Ch, Co, Ev, Me, Re, Us>,
+) => {
+  const {
+    disabled: prevDisabled,
+    giphyActive: prevGiphyActive,
+    sendMessage: prevSendMessage,
+  } = prevProps;
+  const {
+    disabled: nextDisabled,
+    giphyActive: nextGiphyActive,
+    sendMessage: nextSendMessage,
+  } = nextProps;
+  const disabledEqual = prevDisabled === nextDisabled;
+  if (!disabledEqual) return false;
+  const giphyActiveEqual = prevGiphyActive === nextGiphyActive;
+  if (!giphyActiveEqual) return false;
+  const sendMessageEqual = prevSendMessage === nextSendMessage;
+  if (!sendMessageEqual) return false;
+  return true;
+};
+const MemoizedSendButton = React.memo(
+  SendButtonWithContext,
+  areEqual,
+) as typeof SendButtonWithContext;
+export type SendButtonProps<
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+> = Partial<SendButtonPropsWithContext<At, Ch, Co, Ev, Me, Re, Us>>;
+/**
+ * UI Component for send button in MessageInput component.
+ */
+export const NewSendButton = (
```

Comment:
Want to name this so it's the same as the file? I just renamed it all randomly for the example last night going fast, better to keep it consistent with how we do the rest.